connector/connect/server/src/test/scala/org/apache/spark/sql/connect/service/SparkConnectSessionHodlerSuite.scala

```diff
@@ -79,4 +90,135 @@ class SparkConnectSessionHolderSuite extends SharedSparkSession {
       sessionHolder.getDataFrameOrThrow(key1)
     }
   }
+
+  private def dummyPythonFunction(sessionHolder: SessionHolder): SimplePythonFunction = {
+    // Needed because pythonPath in PythonWorkerFactory doesn't include test spark home
+    val sparkPythonPath = PythonUtils.mergePythonPaths(
+      Seq(sparkHome, "python", "lib", "pyspark.zip").mkString(File.separator),
+      Seq(sparkHome, "python", "lib", PythonUtils.PY4J_ZIP_NAME).mkString(File.separator))
+
+    SimplePythonFunction(
+      command = Array.emptyByteArray,
+      envVars = mutable.Map("PYTHONPATH" -> sparkPythonPath).asJava,
+      pythonIncludes = sessionHolder.artifactManager.getSparkConnectPythonIncludes.asJava,
+      pythonExec = IntegratedUDFTestUtils.pythonExec,
+      pythonVer = IntegratedUDFTestUtils.pythonVer,
+      broadcastVars = Lists.newArrayList(),
+      accumulator = null)
+  }
+
+  test("python foreachBatch process: process terminates after query is stopped") {
+    val sessionHolder = SessionHolder.forTesting(spark)
```

Comment:
How does this test if the actual python process is not running?

Reply:
Found out a new way of testing this. Please see the update